drivers/hmis/spec/models/hmis/hmis_data_cleanup_util_spec.rb

```diff
@@ -0,0 +1,222 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'rails_helper'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe HmisDataCleanup::Util, type: :model do
+  let!(:hmis_ds) { create :hmis_data_source }
+  let!(:o1) { create :hmis_hud_organization, data_source: hmis_ds }
+  let!(:p1) { create :hmis_hud_project, data_source: hmis_ds, organization: o1 }
+  let(:last_year) { 1.year.ago }
+  let!(:e1) { create :hmis_hud_enrollment, DateCreated: last_year, DateUpdated: last_year, data_source: hmis_ds, project: p1 }
+  let!(:e2) { create :hmis_hud_enrollment, DateCreated: last_year, DateUpdated: last_year, data_source: hmis_ds, project: p1 }
+  let!(:e3) { create :hmis_hud_enrollment, DateCreated: last_year, DateUpdated: last_year, data_source: hmis_ds }
+  let!(:e4) { create :hmis_hud_enrollment, DateCreated: last_year, DateUpdated: last_year, data_source: hmis_ds }
+
+  before(:all) do
+    # create cruft in other data sources
+    [:destination_data_source, :source_data_source].each do |ds_factory|
+      ds = create(ds_factory)
+      proj = create(:hud_project, data_source_id: ds.id)
+      client = create(:grda_warehouse_hud_client, data_source_id: ds.id)
+      5.times do
+        create(:grda_warehouse_hud_enrollment, data_source_id: ds.id, client: client, project: proj, DateCreated: 1.year.ago, DateUpdated: 1.year.ago)
+      end
+    end
+  end
+
+  def expect_leaves_non_hmis_data_alone(&block)
+    expect do
+      yield block
+    end.to change(GrdaWarehouse::Hud::Enrollment, :count).by(0).
+      and change(GrdaWarehouse::Version, :count).by(0).
+      # none of the utils should set timestamps on any records
+      and change(GrdaWarehouse::Hud::Enrollment.where(DateUpdated: 5.minutes.ago..), :count).by(0).
+      and change(GrdaWarehouse::Hud::Service.where(DateUpdated: 5.minutes.ago..), :count).by(0).
+      and change(GrdaWarehouse::Hud::CurrentLivingSituation.where(DateUpdated: 5.minutes.ago..), :count).by(0).
+      and change(GrdaWarehouse::Hud::Client.where(DateUpdated: 5.minutes.ago..), :count).by(0)
+  end
+
+  context 'clear_enrollment_export_ids' do
+    it 'works' do
+      GrdaWarehouse::Hud::Enrollment.update_all(ExportID: 'XYZ')
```

Comment:
minor style suggestion for these context blocks: maybe move shared context into a before each block ( also repeated in 'leaves no trace')

```suggestion
  context 'enrollments with export ids' do
    before(:each)  { GrdaWarehouse::Hud::Enrollment.update_all(ExportID: 'XYZ') }
    it 'clears the exports' do
```

Reply:
thank you! always appreciate style suggestions on rspec, somehow nothing I write in rspec ever feels neat.